cli/remotebazel/remotebazel.go

```diff
@@ -683,14 +695,18 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	healthChecker := healthcheck.NewHealthChecker("remote-bazel-client")
 	env := real_environment.NewRealEnv(healthChecker)
 
+	ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-api-key", opts.APIKey)
+
+	// Handle interrupts to cancel the remote run.
+	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
+	defer cancel()
```

Comment:
should we move this after the `Run()` RPC to ensure that we cancel the remote invocation if it has been scheduled? (I think the previous code was doing that)

Reply:
I had put it before the `Run` RPC because if it takes a while for the runner to get scheduled and you try to CTRL-C, it will not be registered and the script will still continue. Updated to use `context.WithoutCancel`

Reply:
Actually the more I think about that case (remote runner takes a long time to get scheduled), it feels incorrect that we would wait until after the runner has been scheduled to cancel the executions it has created (this could take a long time if there is queueing). It seems like we should just cancel that remote run request immediately. Though I acknowledge there could be a race condition where the remote runner has created the invocation and we do not cancel it

Reply:
Actually the more I think about that case (remote runner takes a long time to get scheduled), it feels incorrect that we would wait until after the runner has been scheduled to cancel the executions it has created (this could take a long time if there is queueing). It seems like we should just cancel that remote run request immediately. Though I acknowledge there could be a race condition where the remote runner has created the invocation and we do not cancel it

I changed it back to previous behavior - where Run takes a cancelable context, but let me know if you still disagree

Reply:
If we ran with `async` and had the run request return the `execution_id` instead of just `invocation_id`, then it seems like we could probably figure out a way to cancel the execution based on the execution_id rather than the invocation ID?

Reply:
> I acknowledge there could be a race condition where the remote runner has created the invocation and we do not cancel it

To address this part more directly - I think we should try to avoid this race condition, since it's important for cancellation to work reliably. e.g. if you're running a `bazel run` that does a deployment, you want to be sure that it's cancelled if you press Ctrl+C

Reply:
Rather than have the remote bazel handler handle the case described above, I added some code in the hostedrunner to cancel the invocation if the context is canceled. Given the point you made above, this seemed like the correct behavior given that some customers are directly using the Run API